### ISSUE TITLE
panoramic images tested is poor

### ISSUE THREAD

**Issue by hcscysh** [ultralytics/ultralytics#22657]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I trained my YOLOv11 on a panoramic images, but the generalization performance in the tests was poor. I'd like to know if I need to adjust my training strategy or the training code.Below is my training code.

import warnings
warnings.filterwarnings('ignore')
from ultralytics import YOLO

if __name__ == '__main__':
    model = YOLO('yolov11_checkpoints/yolo11x-obb.pt')
    model.train(
        data='datasets/Cultural_Relics_Detection.yaml',  # path to dataset YAML
        epochs=500,  # number of training epochs
        batch=8,
        imgsz=640,  # training image size 640
        device="0,1",  # device to run on, i.e. device=0 or device=0,1,2,3 or device=cpu
    )

**Comment by UltralyticsAssistant** [comment]
👋 Hello @hcscysh, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get moving quickly—an Ultralytics engineer will also assist you soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including:
- Dataset details (panoramic image examples, annotation screenshots, class distribution)
- Exact training command/code, full logs, and results summary (metrics, losses)
- Validation settings (imgsz, augmentations, rect/mosaic usage) and any changes from defaults
- A brief note on image aspect ratios and whether you used rect training for panoramas
And verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). Note: current models are named YOLO11 (e.g., `yolo11x-obb.pt`) 😊

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you post labels.png from your runs folder?

**Comment by hcscysh** [comment]
@Y-T-G 

![Image](https://github.com/user-attachments/assets/f136c8cc-3ec5-4f27-b2cc-2f9d630fdb87)

**Comment by Y-T-G** [comment]
Your datasst is imbalanced and your objects are too small. You should use larger image size. And balance the classes

**Comment by hcscysh** [comment]
@Y-T-G 
Thank you for your reply. I changed the size from 640 to 2560. Since the dataset's annotations are imbalanced, is it possible to balance the dataset using Ultralytics' built-in code?

**Comment by Y-T-G** [comment]
You can try this

https://y-t-g.github.io/tutorials/yolo-class-balancing/

**Comment by thi23459112** [comment]
> [@Y-T-G](https://github.com/Y-T-G) Thank you for your reply. I changed the size from 640 to 2560. Since the dataset's annotations are imbalanced, is it possible to balance the dataset using Ultralytics' built-in code?

Just wanted to check whether adjusting the imgsz improved your results. I’m also working with panoramic images, where some objects appear very large while others are quite small.

**Comment by thi23459112** [comment]
@Y-T-G 
I’d like to ask for some advice. Since the image is a panoramic photo, the resolution is very large. Some objects take up more than 50% of the frame, while others are so small that they’re only a few dozen pixels. How can I improve detection performance in this kind of situation? I’m already using yolo11x-p2.yaml, but the results still aren’t great.
Do you have any suggestions or ideas?

**Comment by Y-T-G** [comment]
You can use SAHI

https://docs.ultralytics.com/guides/sahi-tiled-inference/

**Comment by thi23459112** [comment]
> You can use SAHI
> 
> https://docs.ultralytics.com/guides/sahi-tiled-inference/

However, when using SAHI, it tends to mishandle some large objects, which increases misdetections.

**Comment by glenn-jocher** [comment]
@fcu52005505 SAHI can indeed hurt large-object detection if those objects are being cut across slices, so you’ll want to tune it in that direction: increase `slice_height`/`slice_width` so most large objects fit inside a single tile, bump `overlap_height_ratio`/`overlap_width_ratio` (e.g. around 0.3–0.5) so borders are well covered, and slightly relax confidence/NMS so overlapping large detections are merged instead of dropped; for extreme scale variation it also helps to train on tiled crops (not just infer on tiles) so the model actually learns from high‑res small objects, as described in the image-tiling section of our [evaluation and fine-tuning guide](https://docs.ultralytics.com/guides/model-evaluation-insights/).

**Comment by thi23459112** [comment]
@Y-T-G @glenn-jocher 
In my dataset, some objects occupy as much as 70–80% of the image. If I apply SAHI directly, those large objects are very likely to get cut across tiles, which can actually hurt detection performance.
On top of that, I’m trying to train a general-purpose model where the same scene can contain both very large and very small objects, rather than a fixed-view setting where objects are consistently small (which is where SAHI really shines).
Because of this, SAHI is not necessarily a better choice for my use case.

**Comment by glenn-jocher** [comment]
You’re right that SAHI isn’t always ideal for scenes with such extreme scale variation; in this case I’d keep standard high‑`imgsz` training/inference so very large objects are seen whole, and augment your dataset offline with extra high‑resolution crops around small objects so a single YOLO11 model learns both global and local views. When doing this, also double‑check class balance and the proportion of small objects as outlined in the [troubleshooting common YOLO issues guide](https://docs.ultralytics.com/guides/yolo-common-issues/).
